### PR TITLE
added dropdown items

### PR DESCRIPTION
**Pull request by ingawei**
image

### CODE REVIEW

File: web/hooks/use-follows.ts [hunk 1]
```diff
@@ -31,9 +31,11 @@ export const useFollowers = (userId: string | undefined) => {
...
-export const useContractFollows = (contractId: string) => {
+export const useContractFollows = (contractId: string | undefined) => {
   const [followIds, setFollowIds] = useState<string[] | undefined>()
-
+  if (!contractId) {
+    return
+  }
   useEffect(() => {
     return listenForContractFollows(contractId, setFollowIds)
   }, [contractId])
```
Comment (jahooma) [inline web/hooks/use-follows.ts:41]: Hmm, the early return would violate one of the rules of hooks: every code path must evaluate all the hooks.

https://reactjs.org/docs/hooks-rules.html

I would instead add an `if` within the `useEffect` to make sure `contractId` is defined.

File: web/components/notifications/notification-dropdown.tsx [hunk 2]
```diff
@@ -73,15 +77,46 @@ function useNotificationPreferenceItem(notification: Notification) {
...
+
+function useNotificationFollowItem(notification: Notification) {
+  const sourceContractId = notification.sourceContractId
+  const sourceContractSlug = notification.sourceContractSlug
+  const followers = useContractFollows(sourceContractId)
+  const user = useUser()
+  if (!user || !sourceContractId || !sourceContractSlug || !followers) {
+    return []
+  }
+  const isFollowing = followers?.includes(user.id)
```
Comment (jahooma) [inline web/components/notifications/notification-dropdown.tsx:100]: `followers` is confusing if it refers to the contractIds you are watching.

Wait, I think `isFollowing` should be checking if your contract follows includes `sourceContractId`, not `user.id`

File: web/components/buttons/follow-market-button.tsx [hunk 3]
```diff
@@ -84,3 +72,31 @@ export const FollowMarketButton = (props: {
     </Tooltip>
   )
 }
+
+export async function unfollowMarket(
+  contractid: string,
+  contractslug: string,
```
Comment (jahooma) [inline web/components/buttons/follow-market-button.tsx:78]: I'd use camelCase here.

**Review by jahooma — APPROVED** [review]
Thanks for quickly building this, Inga! Our users will love it.

Just a couple comments for you to address.

File: web/components/notifications/notification-dropdown.tsx (hunk 2 above)
Comment (ingawei) [reply web/components/notifications/notification-dropdown.tsx:100]: is that right? I thought followers is the list of people following a contract, so you're looking for yourself in that list

Comment (ingawei) [reply web/components/notifications/notification-dropdown.tsx:100]: yee it's how FollowMarketButton uses it
```
export const FollowMarketButton = (props: {
  contract: Contract
  user: User | undefined | null
}) => {
  const { contract, user } = props
  const followers = useContractFollows(contract.id)
  const [open, setOpen] = useState(false)

  const watching = followers?.includes(user?.id ?? 'nope')
```

Comment (jahooma) [reply web/components/notifications/notification-dropdown.tsx:100]: Oh, you were right

**Pull request merged**